### PR TITLE
Validate rbd image name in NodeExpand

### PR DESCRIPTION
**Pull request by Madhu-1**
Added one more validation in NodeExpand for the volume name from the request and the image name stored in the staging path metadata.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: pkg/rbd/nodeserver.go [hunk 1]
```diff
@@ -553,16 +553,19 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
...
-
-	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
-	devicePath, err := getDevicePath(ctx, volumePath)
+	imgInfo, devicePath, err := getDevicePathAndImageInfo(ctx, volumePath)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+	// comapare volume name from request and volume name stored in stash
+	if volName != imgInfo.ImageName {
+		return nil, status.Errorf(codes.InvalidArgument, "volume name miss match between request (%s) and stored metadata (%s)", volName, imgInfo.ImageName)
```
Comment (ShyamsundarR) [inline pkg/rbd/nodeserver.go]: (nit) "mismatch" is a single word.

Comment (Madhu-1) [reply pkg/rbd/nodeserver.go]: done

**Comment by Madhu-1** [comment]
@humblec @ShyamsundarR PTAL

**Comment by humblec** [comment]
LGTM.

File: pkg/rbd/nodeserver.go [hunk 2]
```diff
@@ -553,16 +553,19 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
...
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-
-	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
-	devicePath, err := getDevicePath(ctx, volumePath)
+	imgInfo, devicePath, err := getDevicePathAndImageInfo(ctx, volumePath)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+	// comapare volume name from request and volume name stored in stash
```
Comment (nixpanic) [inline pkg/rbd/nodeserver.go]: Nit: "comapare" -> "compare"

It is also pretty clear what is checked, there it not really a need to explain that :-)

**Pull request merged**